Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,370 @@
+
+The document supplements canonical Bazel resources:
+- [Bazel overview](https://docs.bazel.build/versions/master/bazel-overview.html)
+- [A tutorial for iOS apps](https://docs.bazel.build/versions/master/tutorial/ios-app.html) 
+- [Bazel Concepts and terminology](https://docs.bazel.build/versions/master/build-ref.html)
+- [Getting started with Bazel](https://docs.bazel.build/versions/master/getting-started.html) 
+
+## Build systems for iOS developers
+
+First, let's address the definiton of a `Build system`. According to [Stack
+overflow](https://stackoverflow.com/questions/7249871/what-is-a-build-tool_),
+_Build tools are programs that automate the creation of executable applications
+from source code(eg. .apk for android app). Building incorporates
+compiling,linking and packaging the code into a usable or executable form._
+
+In iOS development, Apple encapsulates build systems inside of Xcode. Xcode is
+both the IDE _and build system_. For many developers, Xcode just works. To have
+a good experience, most developers don't need to worry about implementing build
+systems, IDEs, or compilers. When the project scales, developer experience
+drops off. Tasks like building, indexing, or merging a change that adds a new
+file can become painful. 
+
+Bazel makes it easy to manage build configuration in a way that results are
+functional and reproducible. In addition to producing an iOS application, Bazel
+makes it easy to automate many other kinds tasks for example, code generating a
+thrift schema, generating an Xcode project, and pushing a docker container.
+Having the ability to optimize the build system can significantly improve the
+developer experience.
+
+Compared to other mainstream build systems, Bazel is strict about inputs and
+outputs. This characteristic makes it reproducible and well suited for
+distributed cloud builds and caching. _To learn about how Bazel compares to
+other build systems, Microsoft's paper [Build Systems à la
+Carte](https://www.microsoft.com/en-us/research/uploads/prod/2018/03/build-systems.pdf),
+compares popular build systems._
+
+## Introduction to Bazel projects
+
+The root of a Bazel project contains two human readable files:
+
+The `BUILD` file - these files define what targets are inside of a project.
+Applications, extensions, static libraries, and frameworks are all declared in
+BUILD files.
+
+The `WORKSPACE` file - this file is about getting files and dependencies from
+[outside the world into
+Bazel](https://docs.bazel.build/versions/master/be/workspace.html).  Simply
+put, external-to-bazel dependencies are put here. 
+
+_The closest notion to BUILD files in the Xcode world, is the machine readble
+`xcodeproj` files managed by Xcode._
+
+## WORKSPACE configuration and setting up rules_apple
+
+For building iOS applications, most iOS developers use the rule set
+`rules_apple`. `rules_apple` contains key rules for iOS development which
+includes rules to build applications, unit tests, and more.
```

Comment:
Might be helpful to either describe or link to Bazel's documentation of rules here (what is a rule? etc)

Reply:
Good point, will link to the latter segment:
https://github.com/pinterest/xchammer/pull/235/files#diff-af23f534f9648f42a375a8dbf3792f8bR136